app/services/idv/steps/upload_step.rb

```diff
@@ -24,28 +39,29 @@ def application
         identity&.friendly_name || 'login.gov'
       end
 
-      def mobile_to_desktop
+      def send_user_to_email_sent_step
         mark_step_complete(:send_link)
         mark_step_complete(:link_sent)
         UserMailer.doc_auth_desktop_link_to_sp(
           current_user, current_user.email, application, link
         ).deliver_later
       end
 
-      def desktop
-        mark_step_complete(:send_link)
-        mark_step_complete(:link_sent)
+      def send_user_to_send_link_step
         mark_step_complete(:email_sent)
-        mark_step_complete(:mobile_front_image)
-        mark_step_complete(:mobile_back_image)
       end
 
-      def mobile
+      def bypass_send_link_steps
         mark_step_complete(:send_link)
         mark_step_complete(:link_sent)
         mark_step_complete(:email_sent)
-        mark_step_complete(:front_image)
-        mark_step_complete(:back_image)
+      end
+
+      def mobile_device?
+        @mobile_device ||= begin
+          client = DeviceDetector.new(request.user_agent)
+          client.device_type != 'desktop'
+        end
```

Comment:
memoizing on a value that could be false means we have to potentially do the work twice.

If we're worried about performance, we can do:

```suggestion
        return @mobile_device if defined?(@mobile_device)
        
        client = DeviceDetector.new(request.user_agent)
        @mobile_device = client.device_type != 'desktop'
```

Reply:
I'm not sure how frequently this gets called, but DeviceDetector can be quite slow so I think it'd be good to fix the memoization